torch/signal/windows/windows.py

```diff
@@ -0,0 +1,315 @@
+    """,
+    r"""
+
+Args:
+    {M}
+    center (float, optional): where the center of the window will be located.
+        In other words, at which sample the peak of the window can be found.
+        Default: `M / 2` if `sym` is `False`, else `(M - 1) / 2`.
+    tau (float, optional): the decay value. Default: 1.0.
+    {sym}
+
+.. note::
+    The window is normalized to 1 (maximum value is 1), however, the 1 doesn't appear if `M` is even
+    and `sym` is `True`.
+
+Keyword args:
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Examples:
+    >>> # Generate an exponential window without keyword args.
+    >>> torch.signal.windows.exponential(10)
+    tensor([0.0111, 0.0302, 0.0821, 0.2231, 0.6065, 0.6065, 0.2231, 0.0821, 0.0302,
+        0.0111])
+
+    >>> # Generate a periodic exponential window and decay factor equal to .5
+    >>> torch.signal.windows.exponential(10,sym=False,tau=.5)
+    tensor([4.5400e-05, 3.3546e-04, 2.4788e-03, 1.8316e-02, 1.3534e-01, 1.0000e+00,
+        1.3534e-01, 1.8316e-02, 2.4788e-03, 3.3546e-04])
+    """.format(
+        **window_common_args
+    ),
+)
+def exponential(
+        M: int,
+        center: float = None,
+        tau: float = 1.0,
+        sym: bool = True,
+        *,
+        dtype: torch.dtype = None,
+        layout: torch.layout = torch.strided,
+        device: torch.device = None,
+        requires_grad: bool = False
+) -> Tensor:
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    _window_function_checks('exponential', M, dtype, layout)
+
+    if M == 0:
+        return torch.empty((0,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if M == 1:
+        return torch.ones((1,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if tau <= 0:
+        raise ValueError(f'Tau must be positive, got: {tau} instead.')
+
+    if not sym and center is not None:
+        raise ValueError('Center must be \'None\' for non-symmetric windows')
+
+    if center is None:
+        center = -(M if not sym else M - 1) / 2.0
+
+    constant = 1 / tau
+
+    """
+    Note that non-integer step is subject to floating point rounding errors when comparing against end;
+    thus, to avoid inconsistency, we added an epsilon equal to `step / 2` to `end`.
+    """
+    k = torch.arange(start=center * constant,
+                     end=(center + (M - 1)) * constant + constant / 2,
+                     step=constant,
+                     dtype=dtype,
+                     layout=layout,
+                     device=device,
+                     requires_grad=requires_grad)
+
+    return torch.exp(-torch.abs(k))
+
+
+@_add_docstr(
+    r"""
+Computes a window with a simple cosine waveform.
+Also known as the sine window.
+
+The cosine window is defined as follows:
+
+.. math::
+    w(n) = \cos{\left(\frac{\pi n}{M} - \frac{\pi}{2}\right)} = \sin{\left(\frac{\pi n}{M}\right)}
+
+Where `M` is the length of the window.
```

Comment:
I would remove this sentence as `M` is defined in the next sentence in the args.